packages/agreement-process/src/services/agreementSuspensionProcessor.ts

```diff
@@ -124,9 +123,21 @@ export async function suspendAgreementLogic({
     ...updateSeed,
   };
 
-  return toCreateEventAgreementUpdated(
-    updatedAgreement,
-    agreement.metadata.version,
-    correlationId
-  );
+  if (authData.organizationId === agreement.data.consumerId) {
+    return toCreateEventAgreementSuspendedByConsumer(
+      updatedAgreement,
+      agreement.metadata.version,
+      correlationId
+    );
+  } else if (authData.organizationId === agreement.data.producerId) {
+    return toCreateEventAgreementSuspendedByProducer(
+      updatedAgreement,
+      agreement.metadata.version,
+      correlationId
+    );
+  } else {
```

Comment:
@MalpenZibo / @beetlecrunch / @galales question after reviewing the updates to suspend agreement tests: there is a case where the consumer is also the producer. In this case, both flags are computed to `true`, and the agreement will become both `suspendedByProducer` and `suspendedByConsumer`. With this code, we correctly update both flags in the agreement but always trigger a `SuspendedByConsumer` event. Is it ok? (same thing for agreement activation)

Reply:
Uhmmm nice catch!
A similar case is also present in the suspension of purposes.
In that case we handle the suspension as producer and I would use the same approach here.
@galales, what do you think?

Reply:
I changed the order of the two if clauses to give the producer event priority (here and also for the activation event)

Reply:
I changed the order of the two if clauses to give the producer event priority (here and also for the activation event)